pkg/kubelet/kubelet_test.go

```diff
@@ -4116,5 +4116,73 @@ func TestCrashLoopBackOffConfiguration(t *testing.T) {
 			assert.Equalf(t, tc.expectedInitial, resultInitial, "wrong base calculated, want: %v, got %v", tc.expectedInitial, resultInitial)
 		})
 	}
+}
+
+func TestSyncPodWithErrorsDuringInPlacePodResize(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+
+	pod := podWithUIDNameNsSpec("12345678", "foo", "new", v1.PodSpec{
+		Containers: []v1.Container{
+			{Name: "bar"},
+		},
+	})
+
+	// Pod resize error returned from the runtime should be surfaced in the conditions.
+	testKubelet.fakeRuntime.SyncResults = &kubecontainer.PodSyncResult{
```

Comment:
Maybe add another test case where the SyncResults are an error, but the resize is successful?

Reply:
added!